daft/scarf_telemetry.py

```diff
@@ -0,0 +1,92 @@
+import platform
+import urllib.parse
+import urllib.request
+from typing import Union
+
+from daft import get_build_type, get_version
+
+
+def scarf_telemetry(scarf_opt_out: bool, runner: str) -> tuple[Union[str, None], Union[str, None]]:
+    """Track analytics for Daft usage via Scarf.
+
+    Args:
+        user_opted_out (bool): Whether the user has opted out of analytics
+        runner (str): The runner being used (py, ray, or native)
+
+    Returns:
+        tuple[str | None, str | None]: Response status and runner type, or (None, None) if analytics disabled/failed
+    """
+    version = get_version()
+    build_type = get_build_type()
+
+    try:
+        # Skip analytics for dev builds or if user opted out
+        if build_type == "dev" or scarf_opt_out:
+            return None, None
```

Comment:
i'd move this out of the `try`
```suggestion
    build_type = get_build_type()
    # Skip analytics for dev builds or if user opted out
    if build_type == "dev" or scarf_opt_out:
        return
    try:

```